src/Simplex/Chat.hs

```diff
@@ -1312,7 +1312,45 @@ processChatCommand = \case
     case conn'_ of
       Just conn' -> pure $ CRConnectionIncognitoUpdated user conn'
       Nothing -> throwChatError CEConnectionIncognitoChangeProhibited
+  APICheckConnectionRequestURI userId (ACR SCMInvitation cReq) -> do
+    -- TODO [repeat connect]
+    -- - search connections (getConnectionEntity?) where conn_req_inv = cReq (and user_id? if yes add to idx_connections_conn_req_inv)
+    --   - if connection not found, return CRUInvitationLinkOkToConnect
+    --   - if connection entity is not RcvDirectMsgConnection, throw chat error prohibited (?)
+    --   - if connection found and contactConnInitiated, return CRUInvitationLinkOwn
+    --   - if connection found and not ready, return CRUInvitationLinkConnecting
+    --   - if connection found and ready, return CRUInvitationLinkKnown (? - see ConnReqURICheckResult comment)
+    ok_
+  APICheckConnectionRequestURI userId (ACR SCMContact cReq) -> do
+    -- TODO [repeat connect]
+    -- - check if cReq is contact address or group link by parsing cReq crClientData (see connectViaContact):
+    --   let groupLinkId = crClientData >>= decodeJSON >>= \(CRDataGroup gli) -> Just gli
+    -- - if contact address:
+    --   - search user_contact_links by conn_req_contact = cReq (and user_id?)
+    --     - if found, return CRUContactAddressOwn
+    --   - search connections by via_contact_uri_hash = hash(cReq) (and user_id?)
+    --     - if connection not found, return CRUContactAddressOkToConnect
+    --     - if connection found and not ready, return CRUContactAddressConnecting
+    --     - if connection found and ready, return CRUContactAddressKnown
+    -- - if group link *:
+    --   - search user_contact_links by conn_req_contact = cReq (and user_id?)
+    --     - if found, return CRUGroupLinkOwn
+    --   - search connections and groups ** by via_contact_uri_hash (via_group_link_uri_hash) = hash(cReq) (and user_id?)
+    --     - if group not found, return CRUGroupLinkOkToConnect ***
+    --     - if connection found and not ready and group not found, return CRUGroupLinkConnecting
+    --     - if group found, return CRUGroupLinkKnown (don't chek membership? or check !active and !removed?)
+    --   (*) rework group links to connect directly to group, w/t creating contact? would remove merge scenario
+    --   (**) host contact may be deleted, but group may exist - no reason to reconnect with contact;
```

Comment:
is it then matchable if host is deleted?

Reply:
via_contact_uri_hash should be persisted to group (for example as via_group_link_uri_hash)